proto/interchain_security/ccv/provider/v1/tx.proto

```diff
@@ -61,3 +63,29 @@ message MsgSubmitConsumerDoubleVoting {
 }
 
 message MsgSubmitConsumerDoubleVotingResponse {}
+
+message MsgOptIn {
+  option (gogoproto.equal) = false;
+  option (gogoproto.goproto_getters) = false;
+  // the chain id of the consumer chain to opt in to
+  string chain_id = 1;
+  // the validator address on the provider
+  string provider_addr = 2 [ (gogoproto.moretags) = "yaml:\"address\"" ];
+  // (optional) the consensus public key to use on the consumer in json string format corresponding to proto-any,
+  // for example `{"@type":"/cosmos.crypto.ed25519.PubKey","key":"Ui5Gf1+mtWUdH8u3xlmzdKID+F3PK0sfXZ73GZ6q6is="}`
+  // we can set `consumer_key = ""` if we do not consider the `consumer_key`
+  string consumer_key = 3;
```

Comment:
Could not set as `optional` field because:
```
Generating Protobuf files
Generating gogo proto code
Warning: plugin "gocosmos" does not support required features.
  Feature "proto3 optional" is required by 1 file(s):
    interchain_security/ccv/provider/v1/tx.proto
Warning: plugin "grpc-gateway" does not support required features.
  Feature "proto3 optional" is required by 1 file(s):
    interchain_security/ccv/provider/v1/tx.proto
```